src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -0,0 +1,1656 @@
+#endif
+#ifdef SIGGFAULT
+  {  SIGGFAULT,   "SIGGFAULT" },
+#endif
+#ifdef SIGGRANT
+  {  SIGGRANT,    "SIGGRANT" },
+#endif
+  {  SIGHUP,      "SIGHUP" },
+  {  SIGILL,      "SIGILL" },
+#ifdef SIGINFO
+  {  SIGINFO,     "SIGINFO" },
+#endif
+  {  SIGINT,      "SIGINT" },
+#ifdef SIGIO
+  {  SIGIO,       "SIGIO" },
+#endif
+#ifdef SIGIOINT
+  {  SIGIOINT,    "SIGIOINT" },
+#endif
+#ifdef SIGIOT
+// SIGIOT is there for BSD compatibility, but on most Unices just a
+// synonym for SIGABRT. The result should be "SIGABRT", not
+// "SIGIOT".
+#if (SIGIOT != SIGABRT )
+  {  SIGIOT,      "SIGIOT" },
+#endif
+#endif
+#ifdef SIGKAP
+  {  SIGKAP,      "SIGKAP" },
+#endif
+  {  SIGKILL,     "SIGKILL" },
+#ifdef SIGLOST
+  {  SIGLOST,     "SIGLOST" },
+#endif
+#ifdef SIGLWP
+  {  SIGLWP,      "SIGLWP" },
+#endif
+#ifdef SIGLWPTIMER
+  {  SIGLWPTIMER, "SIGLWPTIMER" },
+#endif
+#ifdef SIGMIGRATE
+  {  SIGMIGRATE,  "SIGMIGRATE" },
+#endif
+#ifdef SIGMSG
+  {  SIGMSG,      "SIGMSG" },
+#endif
+  {  SIGPIPE,     "SIGPIPE" },
+#ifdef SIGPOLL
+  {  SIGPOLL,     "SIGPOLL" },
+#endif
+#ifdef SIGPRE
+  {  SIGPRE,      "SIGPRE" },
+#endif
+  {  SIGPROF,     "SIGPROF" },
+#ifdef SIGPTY
+  {  SIGPTY,      "SIGPTY" },
+#endif
+#ifdef SIGPWR
+  {  SIGPWR,      "SIGPWR" },
+#endif
+  {  SIGQUIT,     "SIGQUIT" },
+#ifdef SIGRECONFIG
+  {  SIGRECONFIG, "SIGRECONFIG" },
+#endif
+#ifdef SIGRECOVERY
+  {  SIGRECOVERY, "SIGRECOVERY" },
+#endif
+#ifdef SIGRESERVE
+  {  SIGRESERVE,  "SIGRESERVE" },
+#endif
+#ifdef SIGRETRACT
+  {  SIGRETRACT,  "SIGRETRACT" },
+#endif
+#ifdef SIGSAK
+  {  SIGSAK,      "SIGSAK" },
+#endif
+  {  SIGSEGV,     "SIGSEGV" },
+#ifdef SIGSOUND
+  {  SIGSOUND,    "SIGSOUND" },
+#endif
+#ifdef SIGSTKFLT
+  {  SIGSTKFLT,    "SIGSTKFLT" },
+#endif
+  {  SIGSTOP,     "SIGSTOP" },
+  {  SIGSYS,      "SIGSYS" },
+#ifdef SIGSYSERROR
+  {  SIGSYSERROR, "SIGSYSERROR" },
+#endif
+#ifdef SIGTALRM
+  {  SIGTALRM,    "SIGTALRM" },
+#endif
+  {  SIGTERM,     "SIGTERM" },
+#ifdef SIGTHAW
+  {  SIGTHAW,     "SIGTHAW" },
+#endif
+  {  SIGTRAP,     "SIGTRAP" },
+#ifdef SIGTSTP
+  {  SIGTSTP,     "SIGTSTP" },
+#endif
+  {  SIGTTIN,     "SIGTTIN" },
+  {  SIGTTOU,     "SIGTTOU" },
+#ifdef SIGURG
+  {  SIGURG,      "SIGURG" },
+#endif
+  {  SIGUSR1,     "SIGUSR1" },
+  {  SIGUSR2,     "SIGUSR2" },
+#ifdef SIGVIRT
+  {  SIGVIRT,     "SIGVIRT" },
+#endif
+  {  SIGVTALRM,   "SIGVTALRM" },
+#ifdef SIGWAITING
+  {  SIGWAITING,  "SIGWAITING" },
+#endif
+#ifdef SIGWINCH
+  {  SIGWINCH,    "SIGWINCH" },
+#endif
+#ifdef SIGWINDOW
+  {  SIGWINDOW,   "SIGWINDOW" },
+#endif
+  {  SIGXCPU,     "SIGXCPU" },
+  {  SIGXFSZ,     "SIGXFSZ" },
+#ifdef SIGXRES
+  {  SIGXRES,     "SIGXRES" },
+#endif
+  { -1, NULL }
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// sun.misc.Signal support
+
+void PosixSignals::jdk_misc_signal_init() {
+  // Initialize signal structures
+  ::memset((void*)pending_signals, 0, sizeof(pending_signals));
```

Comment:
Is this code needed? `pending_signals` is initialized with `{ 0 }` and also `jdk_misc_signal_init()` would be called from `os::init_2()` .

Reply:
Will be addressed in https://bugs.openjdk.java.net/browse/JDK-8253742